src/rt/rwlock.rs

```diff
@@ -111,25 +133,6 @@ impl RwLock {
         });
     }
 
-    fn lock_out_threads(&self, execution: &mut Execution, thread_id: thread::Id) {
```

Comment:
The fix required changing the fn details for each call... I opted to re-add some duplication instead of figuring out a way to stay DRY. I didn't think that adding another argument to the fn would be ideal.